frontend/compositions/map/useData.tsx

```diff
@@ -0,0 +1,46 @@
+import { Feature } from "geojson"
+import { useMemo, useState } from "react"
+import uscities from "./uscities.json"
+
+export interface GeoJson {
+  type: string
+  features: Feature[]
+}
+
+export interface Filter {
+  property: string
+  value: number | string
+}
+
+export interface Data {
+  features: Feature[]
+  filter: Filter
+  setFilterProperties: (filter: Filter) => void
+}
+
+export default function useData(): Data {
+  // using temp dummy data set
+  const data: GeoJson = uscities as GeoJson
+
+  const [filterProperties, setFilterProperties] = useState<Filter>({
+    property: "state_name",
+    value: "",
+  })
+
+  return useMemo(() => {
+    const { property, value } = filterProperties
+
+    const filteredFeatures =
+      value === ""
+        ? ([] as Feature[])
+        : value === "all"
+        ? data.features
+        : data.features.filter((d: Feature) => d.properties[property] === value)
```

Comment:
Hmm I feel like this could be simplified as a single ternary (checking if value === 'all' and then returning all or the filtered array) 
And then we could either capture the `value === ''` edge case in the final return statement OR I'm wondering... would `data.features` ever have a feature whose `properties[property]` was an empty string? 
If not - the second part of your second ternary (ie currently line 38) would already capture that case! (ie return an empty array typed as Feature[])

Reply:
Yeah, the temp data set was giving me some trouble filtering on an empty string, but ideally, yeah, we can simplify this like you say.